Mac/README.rst

```diff
@@ -399,8 +399,8 @@ The basic implementation pattern is:
 Resources
 =========
 
-  *  http://www.python.org/download/mac/
+  *  https://www.python.org/downloads/mac-osx/
```

Comment:
Old URL forwards to this.

Reply:
Please don't change this occurence as "mac-osx" is an outdated term. We should rather fix the website to use the modern "macOS" term instead.

Reply:
I think that's best left to a later discussion, this reflects the current state of the world :)

Reply:
The current state of the world is that the redirect works and we don't need to modify the line in the README. If we do, we'll have to change the README again.

Reply:
(if you only want to `s/http/https/` then that's obviously fine)

Reply:
Yes? Changing it again will be correct, unless you want me to just go change the website now, which I can also do :)

Reply:
https://github.com/python/pythondotorg/blob/7556542b58d9c90717757c9fa997637752d6dd85/pydotorg/urls.py#L30 is what's performing the current redirect, but changing this will have deeper impact on a number of views. I understand your point that `mac-osx` is an out of date reference but it's the correct one for python.org.

Reply:
I changed the DB entry on python.org. The URL is now:
https://www.python.org/downloads/macos/

Unfortunately, as you noted, some things are needlessly static in the source code. I created this PR to update it:
https://github.com/python/pythondotorg/pull/1829